### ISSUE TITLE
Chromaticraft reach ability incompatibility with forge essential client.

### ISSUE THREAD

**Issue by jsrfuture** [ReikaKalseki/Reika_Mods_Issues#1103]
having forge essential client make the reach ability not working past the normal 5block.

**Comment by ReikaKalseki** [comment]
Not surprising, and possibly unfixable. Who develops ForgeEssentials?

**Comment by jsrfuture** [comment]
not sure who's active, maybe @luacs1998.

figure i should mention it there, since it took about half an hour to find out why it wasnt working, it might save the trouble to someone else. Even if it might be impossible to fix.

**Comment by jsrfuture** [comment]
quick question, completly unrelated, I've noticed that chromaticraft ability ignore zone protection/permision, like forge essential or mytown2. Do guardian stone protect from those ability, like sonic blast? I don't have anyone to test it with until tomorow.

**Comment by ReikaKalseki** [comment]
Most things should obey permissions where relevant, provided those plugins and their interfaces with Forge are working properly.

**Comment by jsrfuture** [comment]
oh well thanks i'll look into it, love those ability but if they can anihilate a claimed base that gonna a problem. I'm not using cauldron, just forge with a couple server utility mods.

**Comment by ReikaKalseki** [comment]
So this is an ASM conflict. 

Relevant code: https://github.com/ForgeEssentials/ForgeEssentials/blob/af7057744bf009c23de0c333daecad10813de7ca/src/client/java/com/forgeessentials/client/mixin/MixinPlayerControllerMP.java

I have some ideas on how to fix this.

**Comment by jsrfuture** [comment]
Great news, forge essential client is convenient, even if it's not essential.

about ability and protection, only sonic blast seem to still destroy thing and ignore permision, even guardian stone protection. Is it intended or a bug? 

edit : Could it be possible that the sonic blast ability is using ops permision instead of player permision? Will try to remove every ops bypass and test a bit more.

**Comment by ReikaKalseki** [comment]
More likely is that sonic blast is not checking permissions at all. Also, by convenient happenstance, anything that listens to plugins will also obey the Guardian stone.

EDIT: It is ignoring permissions:
https://github.com/ReikaKalseki/ChromatiCraft/blob/master/Registry/Chromabilities.java#L1039

This is very easy to fix.

**Comment by jsrfuture** [comment]
yeah, tested everything i could think of and I don't know java really much. I did try to take a look at this but could understand much of it. It did give me an idea for a temporary work around to protect player base from this. A simple script that will increase block resistance in the claimed area. Not ideal but should do the trick for now, to prevent more "accident".(player A was using sonic blast to collect dirt, player B had an hidden small base that got partially broken)
edit : oh well i dont think it will work finaly, will try to think of something else ><

**Comment by DaMachinator** [comment]
And this should work with _FTBUtils_ as well?

**Comment by Aivech** [comment]
And this _should_ work with FTBUtils as well?

**Comment by ReikaKalseki** [comment]
No idea. My current planned fix is to overwrite FE's changes and then call its code for information - basically extending reach if either FE or CC say to do so - but that would not automatically add FTBUtils compatibility.

**Comment by Aivech** [comment]
FTBUtils has its chunk claiming system. It's hard to describe how it works, but you can claim chunks and share them with others if you so desire.

**Comment by jsrfuture** [comment]
I'm done progressing in chromaticraft. I'm not sure if you'd prefer I'd make a new issue or not, but there is a couple ability that is not checking for permision, including guardian stone.
1. Sonic blast - breaking block ( as mentioned before )
2. Fireball source - fire spread, changing dirt/sand into glass
3. skystrike - breaking block
4. passive draw - interaction with lumen tree 

location warp doesnt seem to work  (I believe it already have been reported)
matter phase seem to ignore entry permision, but this perm is buggy with many other thing so.

**Comment by ReikaKalseki** [comment]
The fire spread in (2) is vanilla mechanics, and (4) only works on Lumen trees you own.

**Comment by jsrfuture** [comment]
wasnt sure if it was possible to do something about (2). About (4) im assuming it was just a visual glitch then.

**Comment by ReikaKalseki** [comment]
I implemented permissions checking for Sonic Blast and Skystrike - the latter only affects blocks, since entities are not bound to block positions.

**Comment by jsrfuture** [comment]
great :) can't wait for the next release.
